dds_md5_reporter.rb

```diff
@@ -4,11 +4,11 @@
 require 'digest'
 
 class DdsMd5Reporter
-  def initialize(upload_id, user_key, agent_key, dds_api_url)
-    @upload_id = upload_id
-    @user_key = user_key
-    @agent_key = agent_key
-    @dds_api_url = dds_api_url
+  def initialize(upload_id:, user_key:, agent_key:, dds_api_url:)
+    @upload_id = upload_id || raise(ArgumentError, "missing keywords: upload_id")
```

Comment:
`ArgumentError` should be raised automatically, I don't think the `|| raise(ArgumentError, ...` part is needed.